website/docs/components/button/partials/guidelines/guidelines.md

```diff
@@ -84,6 +84,43 @@ Buttons used for a Dropdown (with the chevron icon) can be found in [Dropdown](/
   <Hds::Button @color="secondary" @text="No text" @icon="plus" @isIconOnly={{true}} />
 </Hds::ButtonSet>
 
+### Leading
+
+In most cases, use leading icons. Choose icons that add meaning and clarity to the action described in the button’s text.
+
+<Hds::ButtonSet>
+  <Hds::Button @color="secondary" @text="Unlock" @icon="unlock" @iconPosition="leading" />
+  <Hds::Button @color="critical" @text="Delete" @icon="trash" @iconPosition="leading" />
+  <Hds::Button @color="tertiary" @text="Previous" @icon="chevron-left" @iconPosition="leading" />
+</Hds::ButtonSet>
+
+!!! Do
+
+Use the leading position when creating or adding a new object.
+
+<Hds::ButtonSet>
+  <Hds::Button @color="primary" @text="Create variable" @icon="plus" @iconPosition="leading" />
+  <Hds::Button @color="secondary" @text="Add repository" @icon="plus" @iconPosition="leading" />
+</Hds::ButtonSet>
+
+!!!
+
+### Trailing
+
+Consider trailing icons when guiding the user forward through the product or when using the Button for links.
```

Comment:
There are some instances of Buttons with links that don't need an icon at all, like "Cancel", so I would remove the last part to avoid confusion.

```suggestion
Consider trailing icons when guiding the user forward through the product.
```